pages/api/id/[form]/owners.ts

```diff
@@ -74,4 +77,56 @@ export async function activateOrDeactivateFormOwners(
   //A 404 status code for a form Not Found in form_users
   return res.status(404).json({ error: "Form or email Not Found" });
 }
+/**
+ * This method aims to create a unique binding between a form template and a user
+ * who needs to access the form's data later on. Here are some requirements that must be satisfied
+ * in order to associate users to collect a form's data
+ * Rule 1: a valid government email address
+ * Rule 2: the FromID/template must exist
+ * Rule 3: the email must not be already associated to the template
+ * @param req The request object
+ * @param res The response object
+ */
+export async function addEmailToForm(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+  //Get request body
+  const requestBody = req.body ? JSON.parse(req.body) : undefined;
+  //Checkimg the payload's content
+  if (!requestBody?.email || !isValidGovEmail(requestBody.email, emailDomainList.domains)) {
+    return res.status(400).json({ error: "The email is not a valid GC email" });
+  }
+  const formID = req.query.form as string;
+  if (!formID) return res.status(400).json({ error: "Malformed API Request Invalid formID" });
+  //Checking if formID exists in db return true or false
+  //May return true if it exists otherwise false
+  const isFormIDExistResult = await executeQuery(
+    await dbConnector(),
+    "SELECT exists(SELECT 1 FROM templates WHERE id = ($1))",
```

Comment:
Hmmm you don't need this nested query or this query at all infact. It's why we set a unique constraint on the template id and the email :). If you try to create a record and it fails with a unique constraint error then you know the record with the template id and the email already exists. There is also a NOT NULL constraint so if that fails you know the form doesn't exist. In this way we don't need to fire off extra queries and load the database more